pkg/auth/keyring.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package auth
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	defaultSessionName = "_ses.nydus"
+)
+
+type KeyRing struct {
+	lock *sync.RWMutex
+}
+
+func NewKeyRing() *KeyRing {
```

Comment:
Maybe give some unit test cases for this?

Reply:
Agree, please try to add some unit test, especially for keying garbage collection and if the boundary is correctly handled.